pkg/allocation/converters/converter.go

```diff
@@ -132,28 +133,84 @@ func convertLabelSelectorToInternalLabelSelector(in *pb.LabelSelector) *metav1.L
 	return &metav1.LabelSelector{MatchLabels: in.GetMatchLabels()}
 }
 
+func convertGameServerSelectorToInternalGameServerSelector(in *pb.GameServerSelector) *allocationv1.GameServerSelector {
+	if in == nil {
+		return nil
+	}
+	result := &allocationv1.GameServerSelector{
+		LabelSelector: metav1.LabelSelector{MatchLabels: in.GetMatchLabels()},
+	}
+
+	if runtime.FeatureEnabled(runtime.FeatureStateAllocationFilter) {
```

Comment:
Should this also check for `in.GameServerState != nil`?

Reply:
It's not a pointer, the proto generates this struct ([source](https://github.com/markmandel/agones/blob/1ac724d635c27a77081be9f714671f697a7fe48e/pkg/allocation/go/allocation.pb.go#L435-L444)):

```go
// GameServerSelector used for finding a GameServer with matching filters.
type GameServerSelector struct {
	// Labels to match.
	MatchLabels          map[string]string                  `protobuf:"bytes,1,rep,name=matchLabels,proto3" json:"matchLabels,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
	GameServerState      GameServerSelector_GameServerState `protobuf:"varint,2,opt,name=gameServerState,proto3,enum=allocation.GameServerSelector_GameServerState" json:"gameServerState,omitempty"`
	Players              *PlayerSelector                    `protobuf:"bytes,3,opt,name=players,proto3" json:"players,omitempty"`
	XXX_NoUnkeyedLiteral struct{}                           `json:"-"`
	XXX_unrecognized     []byte                             `json:"-"`
	XXX_sizecache        int32                              `json:"-"`
}
```

So no nil check needed.